pkg/envtest/komega/komega.go

```diff
@@ -0,0 +1,260 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package komega
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// komega has Gomega Matchers that use the controller-runtime client.
+type komega struct {
+	ctx          context.Context
+	g            gomega.Gomega
+	client       client.Client
+	extras       []interface{}
+	timeout      time.Duration
+	pollInterval time.Duration
+}
+
+var _ Komega = &komega{}
+
+// NewKomega creates a new instance with a given client.
+func NewKomega(c client.Client) *komega {
```

Comment:
Should this return the exposed interface and not the unexported struct? (same in the funcs below)